Ira/my-app-video/src/components/home/Home.tsx

```diff
@@ -0,0 +1,142 @@
+import React from 'react';
+import {RouteComponentProps} from "react-router-dom";
+import {Header} from "../layout/header";
+import {Footer} from "../layout/footer";
+import {FilmList} from "../filmList";
+import {SortingSection} from "../sortingSection";
+import Pagination from "../pagination";
+import SearchFilmForm from "../shared/searchFilmForm";
+import {queryString} from "../../constants/queryString";
+import {movieConnectProps} from "./index";
+import loader from '../../accets/Dual Ball-1s-200px.svg';
+import {SortBy} from "../sortingSection/SortingSection";
+import './styles.scss';
+
+type HomeProps = RouteComponentProps & movieConnectProps;
+
+export class Home extends React.Component<HomeProps> {
+
+  componentDidMount() {
+    this.getMovies();
+  };
+
+  componentWillUnmount() {
+    this.props.resetMovies();
+  }
+
+  componentDidUpdate(prevProps: Readonly<HomeProps>) {
+    if (this.props.history.action !== 'PUSH' && this.props.location !== prevProps.location) {
+      this.getMovies();
+    }
+  };
+
+  getMovies = () => {
+    const params: { search: string, searchBy: string, sortBy: string, page: number } =
+      queryString.parse(this.props.location.search) as
+        { search: string, searchBy: string, sortBy: string, page: number };
+
+    if (Object.keys(params).length === 0) {
+      this.props.fetchMovies( {sortBy: SortBy.release});
+    } else {
+      this.props.fetchMovies({
+        search: params.search,
+        searchBy: params.searchBy,
+        sortBy: params.sortBy || SortBy.release,
+        page: Number(params.page) || 1,
+      });
+    }
+  };
+
+  onSearchSubmit = ({searchValue, searchBy}: { searchValue: string, searchBy: string}): void => {
+    const params: { search: string, searchBy: string, sortBy: string, page: number } =
+      queryString.parse(this.props.location.search) as
+        { search: string, searchBy: string, sortBy: string, page: number };
+
+    const newParams: string = queryString.stringify({
+      ...params,
+      search: searchValue,
+      searchBy: searchBy,
+    });
+
+    this.props.history.push(`/search?${newParams}`);
+
+    this.props.fetchMovies({
+      search: searchValue,
+      searchBy: searchBy,
+      sortBy: params.sortBy || SortBy.release,
+    });
+  };
+
+  onToggleSortBy = (btnName: string) => {
+    const params: { search: string, searchBy: string, sortBy: string, page: number } =
```

Comment:
I can see that you use {search: string, searchBy: string, sortBy: string} several times in this file. Move it please to a separate interface, so you won't copy paste all the time.